threeML/plugins/Fermi_GBM/download_GBM_data.py

```diff
@@ -38,6 +38,77 @@ def _validate_fermi_trigger_name(trigger):
     return trigger
 
 
+_month2number = {'jan':1,
```

Comment:
You can use the python datetime module to make sure that the requested date is legal. By doing that you can get rid of almost all the checks. You can also make a better check for a date after Fermi launch. Some examples:

```python

In [13]: fermi_data_start = datetime.datetime(2008,8,4,00,00,00)

In [14]: other_date = datetime.datetime(2008,1,1)

In [15]: other_date > fermi_data_start
Out[15]: False

In [16]: other_date_after = datetime.datetime(2009,1,1)

In [17]: other_date_after > fermi_data_start
Out[17]: True

In [18]: generate_error = datetime.datetime(2009,15,27)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-18-be3df6adb69c> in <module>()
----> 1 generate_error = datetime.datetime(2009,15,27)

ValueError: month must be in 1..12

```